### PR TITLE
feat(inpaint): add solid infill for use with inpainting model

### LINKED ISSUE #2417: outpainting option to skip infill method (patchmatch etc)
> I'd like to explore the possibility of special-casing for inpainting models, which do pretty well (if not better) without the second pass used for inpainting and outpainting. Because of this I still keep the ckpt version of inpainting-1.5 around for my outpainting needs because it uses the legacy omnibus module and bypasses the inpainting code.

-- [lstein](https://discord.com/channels/1020123559063990373/1031668022294884392/1066959907829579826)

### Alternatives

We could continue to switch this based on the model, though I'd prefer adding an explicit "None" option to the infill selection options. This would allow comparison of the different infill approaches with different models more directly.

### PR DESCRIPTION
**Pull request by keturn**
A new infill method, **solid:** solid color. currently using middle gray.

Fixes #2417

It seems like the runwayml inpainting model specifically expects those masked areas to be blanked out like this.

I haven't tried the SD 2.0 inpainting model with it yet.

### CODE REVIEW

**Comment by psychedelicious** [comment]
I think it makes sense to expose the color selection to the end user, in which case this should be a parameter.

File: ldm/invoke/generator/inpaint.py [hunk 1]
```diff
@@ -202,12 +204,17 @@ def get_make_image(self,prompt,sampler,steps,cfg_scale,ddim_eta,
...
                 init_filled = self.infill_patchmatch(self.pil_image.copy())
-            else: # if infill_method == 'tile': # Only two methods right now, so always use 'tile' if not patchmatch
+            elif infill_method == 'tile':
                 init_filled = self.tile_fill_missing(
                     self.pil_image.copy(),
                     seed = self.seed,
                     tile_size = tile_size
                 )
+            elif infill_method == 'solid':
+                solid_bg = PIL.Image.new("RGBA", init_image.size, (0x7F, 0x7F, 0x7F, 0xFF))
```
Comment (lstein) [inline ldm/invoke/generator/inpaint.py:214]: I agree with @psychedelicious on this. Could the solid color be exposed as a parameter passed to `get_make_image()`, defaulting to grey?

**Review by lstein — APPROVED** [review]
Other than the request to expose the fill color as a parameter, this looks ready to merge.

File: ldm/invoke/generator/inpaint.py (hunk 1 above)
Comment (lstein) [reply ldm/invoke/generator/inpaint.py]: I implemented the change so that we can move this along.

**Pull request merged**